src/processors/AsyncAPIInputProcessor.ts

```diff
@@ -74,30 +77,80 @@ export class AsyncAPIInputProcessor extends AbstractInputProcessor {
     }
 
     inputModel.originalInput = doc;
-    // Go over all the message payloads and convert them to models
-    for (const message of doc.allMessages()) {
-      const payload = message.payload();
-      if (payload) {
-        const schema = AsyncAPIInputProcessor.convertToInternalSchema(payload);
-        const newCommonModel =
-          JsonSchemaInputProcessor.convertSchemaToCommonModel(schema, options);
-        if (newCommonModel.$id !== undefined) {
-          if (inputModel.models[newCommonModel.$id] !== undefined) {
-            Logger.warn(
-              `Overwriting existing model with $id ${newCommonModel.$id}, are there two models with the same id present?`,
-              newCommonModel
-            );
-          }
-          const metaModel = convertToMetaModel(newCommonModel);
-          inputModel.models[metaModel.name] = metaModel;
-        } else {
+
+    const addToInputModel = (payload: AsyncAPISchemaInterface) => {
+      const schema = AsyncAPIInputProcessor.convertToInternalSchema(payload);
+      const newCommonModel =
+        JsonSchemaInputProcessor.convertSchemaToCommonModel(schema, options);
+
+      if (newCommonModel.$id !== undefined) {
+        if (inputModel.models[newCommonModel.$id] !== undefined) {
           Logger.warn(
-            'Model did not have $id which is required, ignoring.',
+            `Overwriting existing model with $id ${newCommonModel.$id}, are there two models with the same id present?`,
             newCommonModel
           );
         }
+        const metaModel = convertToMetaModel(newCommonModel);
+        inputModel.models[metaModel.name] = metaModel;
+      } else {
+        Logger.warn(
+          'Model did not have $id which is required, ignoring.',
+          newCommonModel
+        );
+      }
+    };
+
+    // Go over all the message payloads and convert them to models
+    const channels = doc.channels();
+
+    if (channels.length) {
+      for (const channel of doc.channels()) {
```

Comment:
In its current setup, Modelina would only generate the message payloads used in operations but not lone survivors under `components.messages`. Is that the intention @kennethaasan ? 🤔

Reply:
`doc.allMessages()` or `operation.messages()` from the parser doesn't return the `oneOf` message in `operation.message.oneOf`. That's why we have to this to include the union type in `operation.message.oneOf`.

Reply:
I wonder if that's not a bug in the parser 🤔 I will check that.

I want to ensure that we generate all the models as before, for all situations otherwise this can quickly become a breaking change. That's why `allMessages()` was a safe choice, cause we didn't have to keep track of all the places where messages could exist 😆 

I think the only thing we are missing is loop through the component messages:
```
for (const message of doc.components().messages()) {
        const payload = message.payload();
        if (payload) {
          addToInputModel(payload);
        }
}
```

Reply:
Please check the parser :)

Do you still want that check for `doc.components().messages()` even though channels exist?

Reply:
In order to not make this a breaking change, then yea, we need to include all those messages 🙂

Reply:
Done :)